nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBRepository.java

```diff
@@ -437,6 +437,23 @@ protected void addIgnoredIds(Document filter, Set<String> ignored) {
         return findAll(filter);
     }
 
+    @Override
+    public List<State> queryKeyValueWithOperator(String key1, Object value1, String key2, DBSQueryOperator operator,
+            Object value2, Set<String> ignored) {
+        Map<String, Object> comparatorAndValue = null;
+        if (DBSQueryOperator.IN == operator) {
```

Comment:
Please use a `switch` for enums.

Reply:
Haha @RSalem07  proposed it but we hesitated (too much)